onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc

```diff
@@ -17,137 +17,126 @@ Y_ZeroPoint = np.clip(round((0 - x_min) / Y_Scale), 0, 255).astype(np.uint8)
-    add_0_to_range.append_binary(dnnl::algorithm::binary_min, zero_mem.get_desc());
-    min_reduction_attr.set_post_ops(add_0_to_range);
+    dnnl::post_ops calc_min;
+    calc_min.append_binary(dnnl::algorithm::binary_min, zero_mem.get_desc());
+    min_reduction_attr.set_post_ops(calc_min);
   }
 
-  auto max_reduction_pd = dnnl::reduction::primitive_desc(max_reduction_d, max_reduction_attr, eng);
-  auto min_reduction_pd = dnnl::reduction::primitive_desc(min_reduction_d, min_reduction_attr, eng);
+  // Create reduction primitive
+  auto max_reduction_prim = dnnl::reduction(dnnl::reduction::primitive_desc(max_reduction_d, max_reduction_attr, eng));
+  auto min_reduction_prim = dnnl::reduction(dnnl::reduction::primitive_desc(min_reduction_d, min_reduction_attr, eng));
 
-  auto max_reduction_prim = dnnl::reduction(max_reduction_pd);
-  auto min_reduction_prim = dnnl::reduction(min_reduction_pd);
+  // Create y_scale and min memory
+  auto y_scale_mem = dnnl::memory(y_scale_md, eng);
+  auto min_reduction_mem = dnnl::memory(y_scale_md, eng);
 
-  auto y_scale_mem = dnnl::memory(min_max_dst_mem_desc, eng);
-  auto min_reduction_dst_mem = dnnl::memory(min_max_dst_mem_desc, eng);
+  // Compute min first since max_reduction needs min as input
+  sp.AddPrimitive(min_reduction_prim, {{DNNL_ARG_SRC, x_mem},
+                                       {DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1, zero_mem},
+                                       {DNNL_ARG_DST, min_reduction_mem}});
 
-  std::unordered_map<int, dnnl::memory> min_reduction_args = {{DNNL_ARG_SRC, x_memory}, {DNNL_ARG_DST, min_reduction_dst_mem}};
-  min_reduction_args[DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1] = zero_mem;
+  // Compute y_scale in fp32
+  sp.AddPrimitive(max_reduction_prim, {{DNNL_ARG_SRC, x_mem}, 
+                                       {DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1, zero_mem},
+                                       {DNNL_ARG_ATTR_MULTIPLE_POST_OP(1) | DNNL_ARG_SRC_1, min_reduction_mem},
+                                       {DNNL_ARG_DST, y_scale_mem}});
 
-  std::unordered_map<int, dnnl::memory> max_reduction_args = {{DNNL_ARG_SRC, x_memory}, {DNNL_ARG_DST, y_scale_mem}};
-  max_reduction_args[DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1] = zero_mem;
-  max_reduction_args[DNNL_ARG_ATTR_MULTIPLE_POST_OP(1) | DNNL_ARG_SRC_1] = min_reduction_dst_mem;
 
-  //compute min first since max_reduction needs min dst as post op arg
-  // compute x min
-  sp.AddPrimitive(min_reduction_prim, min_reduction_args);
-  // compute y scale f32
-  sp.AddPrimitive(max_reduction_prim, max_reduction_args);
+  // y_zero_point = np.clip(round((0 - x_min) / y_scale), 0, 255).astype(np.uint8)
+  // Y_ZERO_POINT COMPUTATION 
+  // Create memory and primitive descriptors
+  auto y_zp_md = dnnl::memory::desc(one_dim, dnnl::memory::data_type::u8, x_format);
+  auto zp_prim_d = dnnl::binary::desc(dnnl::algorithm::binary_div, y_scale_md, y_scale_md, y_zp_md);
```

Comment:
am i reading this correctly you are doing an in place binary div.  i.e. y_scale = y_scale/y_zp?

Reply:
No, this part means that src0 and src1 have the same md (y_scale_md) and the output md is y_zp_md

Reply:
🤦‍♂️ i must have been over thinking it yesterday. I feel kinda silly that I even asked.  Thanks :)